### ISSUE TITLE
Add new users through Chatkit Server

### ISSUE THREAD

**Issue by nyashafrank** [steevejoseph/bazaar-backend#59]
**TL;DR** 
read the whole thing please

Hiii sorry to bother.... I have a somewhat cool chat feature that is super duper close to done, I just need some @backend help.

For this feature, I am using Chatkit by Pusher. Chatkit helps us integrate 1-1 private chats and stores all of our messages for us! On their site, you can create a Chatkit instance and connect to this instance in your app. This instance has the ability to create users that can create public and private chat rooms.

The only problem is that Chatkit wants every user that will be sending and receiving messages through our instance to be added to our instance (programmatically or manually). Right now, my private chat feature only works for the Bazaar user IDs that I have added to the instance manually (through the Chatkit GUI). However, chat needs to be available for ALL Bazaar users.  

To solve this, we would need to add every new user that signs up on Bazaar to our Chatkit instance.

I can explain more when we meet up. But, here are some links for background info. 

- https://pusher.com/chatkit

- https://docs.pusher.com/chatkit/reference/server-node#users

- https://docs.pusher.com/chatkit/authentication

- https://pusher.com/tutorials/typing-indicator-react#set-up-the-server

- https://pusher.com/tutorials/typing-indicator-react#typing-indicator

I know you have a lot of issues on your plate already. So if it's too much work, no worries!

**Comment by jwf1112** [comment]
@nyashafrank 
so just to check you need each user added to the instance/our chat client? and then do you have all the other stuff such as the messaging and logging in to chatkit? 

mainly trying to check if its something as simple as adding more stuff to usersignup and the user model or a whole new set of routes

would probably be good to meet up to get a clearer picture

**Comment by jwf1112** [comment]
a user is now created in our instance upon signup using the user id
